libraries/chain/controller.cpp

```diff
@@ -563,6 +567,13 @@ struct controller_impl {
        for (const auto& abi : abis) {
           restore_contract(snapshot, abi);
        }
+
+       snapshot->read_section<cyberway::chaindb::revision_t>([this]( auto &section ) {
+           cyberway::chaindb::revision_t revision = 0;
+           section.read_row(revision);
+           chaindb.set_revision(revision);
```

Comment:
This change doesn't look good.
`chaindb.set_revision()` is used for replaying, because it skips undo steps.
If it is restoring from the state, the `restore_db()` should be used, which load undo state to the cache and sets revision from undo.

Reply:
This line https://github.com/cyberway/cyberway/blob/master/libraries/chain/controller.cpp#L473 - doesn't work?